src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -3457,6 +3474,19 @@ void EmitSingleCharLazy(RegexNode node, RegexNode? subsequent = null, bool emitL
                                 break;
                         }
                     }
+                    else if (literal.AsciiChars is not null) // set of only ASCII characters
+                    {
+                        char[] asciiChars = literal.AsciiChars;
+                        overlap = asciiChars.AsSpan().Contains(node.Ch);
+                        if (!overlap)
+                        {
+                            Debug.Assert(node.Ch < 128);
+                            Array.Resize(ref asciiChars, asciiChars.Length + 1);
+                            asciiChars[asciiChars.Length - 1] = node.Ch;
```

Comment:
Nit:
```suggestion
                            asciiChars[^1] = node.Ch;
```

Reply:
I think the source gen didn't like this, I'll double-check

Reply:
RegexCompiler.cs shouldn't be included in the source generator.